schema/draft/schema.ts

```diff
@@ -267,6 +267,34 @@ export interface ServerCapabilities {
   };
 }
 
+interface UrlIcon {
+  kind: 'url';
+  /**
+   * A standard URI pointing to an icon resource.
+   * @format uri
+   */
+  src: string;
+  /** Optional override if the server’s MIME type is missing or generic. */
+  mimeType?: string;
+  /** e.g. "48x48", "any" (for SVG), or "48x48 96x96" */
+  sizes?: string;
+}
+
+/** Inline icon: base64-encoded data URI */
+interface DataIcon {
+  kind: 'data';
+  /**
+   * Data URI embedding the image, e.g.:
+   *   "data:image/png;base64,iVBORw0KGgoAAAANS…"
+   */
+  src: `data:${string};base64,${string}`;
+  /** Always required for clarity, though data URI also encodes it */
+  mimeType: string;
+  sizes?: string;
+}
+
+export type Icon = UrlIcon | DataIcon;
```

Comment:
I'm unconvinced a separate icon type between URL or data icons is useful. A data URI is an entirely valid URL and it's easy to distinguish the two by looking at the scheme. Do we see the types here diverging further?

Reply:
+1 to what @connor4312 suggested - it feels like this can be simplified.